### DISCUSSION TITLE
DNN SMTP settings, Mailkit and OAuth2 to send emails via Office365

### DISCUSSION THREAD

**Discussion by edisfera** [dnnsoftware/Dnn.Platform#5179] [General]
Hi,
on the 1st of October 2022 Microsoft is going to discontinue Basic Authentication in favour of OAuth2. This will affect any service that is currently using Office 365 to send "unattended" email. I have already incurred in this issue on one of my services (not DNN) where one day it was working like a charm and the following day stopped working not being able to login O365 to send emails (but fortunately I was still in the grace period and I just re-enabled Basic Authentication on my Tenant to temporarily fix it).
Now, the point is that if one wants DNN to send emails (such as notifications, alerts, etc.) via O365 he/she will no longer able to do so using the current (as far as I know) Mailkit APIs available in DNN. Mailkit itself seems to have limitations in supporting OAuth2 with O365 (https://github.com/jstedfast/MailKit/blob/master/ExchangeOAuth2.md). For sending unattended emails is of course out of question the use of MFA since DNN runs as a service (no user seats in front of that installation ready to insert user, password and codes sents during the multy factor authentication). The only viable solution I have found and tested so far, is to create what Microsoft calls in its documentation a Client Credentials Flow (https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-client-creds-grant-flow) and use Graph APIs to send emails instead of MailKit.
I'd like to know if the comunity is aware of this next to coming issue and if there are already plans to fix it.
Thanks

**Comment by CMarius94** [comment]
Are there any plans to implement this?

https://learn.microsoft.com/en-us/exchange/clients-and-mobile-in-exchange-online/deprecation-of-basic-authentication-exchange-online
Microsoft announced deprecation of SMTP Auth. Even if DNN is not affected (yet) because they will keep SMTP AUTH active with Basic Authentication, it would be nice to implement Modern Authentication (their OAuth2). 

![image](https://user-images.githubusercontent.com/20061507/193073703-9fdc0a08-d737-4b91-bc03-2765d61b2b5b.png)

Also, Microsoft provides this flow to implement and set up SMTP on OAuth2: https://learn.microsoft.com/en-us/exchange/client-developer/legacy-protocols/how-to-authenticate-an-imap-pop-smtp-application-by-using-oauth

**Reply by jasclar** [reply]
see also https://github.com/dnnsoftware/Dnn.Platform/issues/5139

**Comment by Timo-Breumelhof** [comment]
The procedure has been added to DNN docs: https://docs.dnncommunity.org/content/tutorials/server/smtp-servers/set-up-oauth-microsoft-365/index.html
